File: app/src/main/java/vandyke/siamobile/transactionslist/TransactionHolder.java
package vandyke.siamobile.transactionslist;

import android.content.Intent;
import android.net.Uri;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;
import vandyke.siamobile.R;

public class TransactionHolder extends RecyclerView.ViewHolder {

    protected TextView transactionStatus;
    protected TextView transactionId;
    protected TextView transactionValue;

    public TransactionHolder(View itemView) {
        super(itemView);
        transactionStatus = (TextView)itemView.findViewById(R.id.transactionStatus);
        transactionId = (TextView)itemView.findViewById(R.id.transactionHeaderId);
        transactionValue = (TextView)itemView.findViewById(R.id.transactionValue);
        itemView.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                String url = "http://explore.sia.tech/hash.html?hash=" + transactionId.getText().toString().replace("\\s*", "");
                Intent i = new Intent(Intent.ACTION_VIEW);
                i.setData(Uri.parse(url));
                v.getContext().startActivity(i);
            }
        });
        transactionId.setVisibility(View.GONE);
    }
}

File: app/src/main/java/vandyke/siamobile/ui/wallet/transactionslist/TransactionHolder.kt
/*
 * Copyright (c) 2017 Nicholas van Dyke
 *
 * This file is subject to the terms and conditions defined in 'LICENSE.md'
 */

package vandyke.siamobile.ui.wallet.transactionslist

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.TextView
import vandyke.siamobile.R

class TransactionHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    val transactionStatus: TextView = itemView.findViewById(R.id.transactionStatus)
    val transactionId: TextView = itemView.findViewById(R.id.transactionHeaderId)
    val transactionValue: TextView = itemView.findViewById(R.id.transactionValue)
}